test/correctness/intrinsics.cpp

```diff
@@ -37,25 +37,23 @@ void check_saturating_add_sub() {
         int64_t a = min_t + ((max_t - min_t) * i) / N;
         for (int j = 0; j < N; j++) {
             int64_t b = min_t + ((max_t - min_t) * j) / N;
-            Expr add = saturating_add(make_const(halide_t, a), make_const(halide_t, b));
-            Expr sub = saturating_sub(make_const(halide_t, a), make_const(halide_t, b));
-            Expr result_add = simplify(lower_intrinsic(add.as<Call>()));
-            Expr result_sub = simplify(lower_intrinsic(sub.as<Call>()));
-            int64_t reference_add = std::min(std::max(a + b, min_t), max_t);
-            int64_t reference_sub = std::min(std::max(a - b, min_t), max_t);
-            if (!can_prove(result_add == make_const(halide_t, reference_add))) {
-                std::cerr << "failure!\n";
-                std::cerr << "test: " << add << "\n";
-                std::cerr << "result: " << result_add << "\n";
-                std::cerr << "expected: " << reference_add << "\n";
-                abort();
-            }
-            if (!can_prove(result_sub == make_const(halide_t, reference_sub))) {
-                std::cerr << "failure!\n";
-                std::cerr << "test: " << sub << "\n";
-                std::cerr << "result: " << result_sub << "\n";
-                std::cerr << "expected: " << reference_sub << "\n";
-                abort();
+            std::vector<std::pair<Expr, int64_t>> intrinsics_with_reference_answer = {
```

Comment:
Probably best to just use a regular C array here. Constructing/destroying the vector here might be pretty expensive. I had to tweak N to make the test run in a reasonable amount of time.